ITE/11/README.adoc

```diff
@@ -0,0 +1,278 @@
+executing commands or scripts during verification. If invoked via scripts, they
+need to be communicated alongside layouts to the verifier and there are no
+in-toto specific guarantees as to their integrity or if they were issued by the
+authors of the layout. Further, these scripts can be complicated as they need
+to load attestations, once again verify their signatures using keys in the
+layout, before they can actually validate the attributes in question.
+
+=== Example: Verifying Result of a Test Run
+
+The in-toto Attestation Framework has a vetted predicate capturing the results
+of
+link:https://github.com/in-toto/attestation/blob/main/spec/predicates/test-result.md[test runs].
+To verify that the test result attestation indicates a successful test run, the
+following CEL rule can be used.
+
+```yaml
+...
+expectedAttributes:
+    - "predicate.result == 'PASSED'"
+...
+```
+
+=== Example: Verifying SLSA Provenance Attributes
+
+link:https://github.com/in-toto/attestation/blob/main/spec/predicates/provenance.md[SLSA Provenance]
+records various attributes such as the identity of the builder, source
+repository, and various other parameters. These attributes can be verified in
+in-toto layouts as follows.
+
+```yaml
+...
+expectedAttributes:
+    - "predicate.runDetails.builder.id == '<expected value>'"
+    - "predicate.externalParameters.repository == '<expected value>'"
+    - "predicate.externalParameters.ref == '<expected value>'"
+...
+```
+
+=== Example: Verifying Build Isolation via Runtime Trace Attestation
+
+This vetted predicate is used to
+link:https://github.com/in-toto/attestation/blob/main/spec/predicates/runtime-trace.md[record a runtime trace]
+of some supply chain step, such as the build step. It can be used to verify
+that the builder made no network calls. But first, it is necessary to verify
+that the monitor was of an expected type and that its trace policy included
+tracking network calls.
+
+```yaml
+...
+expectedAttributes:
+    - "predicate.monitor.type == 'https://github.com/cilium/tetragon'"
+    - "predicate.monitor.tracePolicy.policies.exists(p, p['Name'] == 'connect')"
+    - "size(predicate.monitorLog.network) == 0"
+...
+```
+
+=== Example: Verifying Contents of Opaque Fields in in-toto Links
+
+As noted above, in-toto links have an opaque `byproducts` field that records
+additional information pertaining to the step performed. in-toto's verification
+workflow does not verify its contents by default. However, the changes proposed
+here can be used to add checks for its contents as well as those of the other
+opaque field, `environment`.
+
+```yaml
+...
+expectedAttributes:
+    - "size(predicate.byproducts['stderr']) == 0"
+    - "predicate.byproducts['return-value'] == 0"
+    - "predicate.environment['<key>'] == '<expected value>'"
+...
+```
+
+[[reasoning]]
+== Reasoning
+
+This ITE presents certain changes that necessitate further reasoning.
+
+=== Use of CEL
```

Comment:
Might also be worth noting that users are not necessarily expected to write raw CEL themselves, but rather higher-level languages could compile down to CEL.